mainwindow.cpp

```diff
@@ -420,15 +431,25 @@ void MainWindow::handleError(QSerialPort::SerialPortError error)
 {
     if (error == QSerialPort::NoError) {
         return;
-    } else if (m_deviceState == DEVICE_OPEN || m_deviceState == DEVICE_OPENING) {
+    } else if (m_deviceState == DEVICE_OPEN || m_deviceState == DEVICE_OPENING || m_deviceState == DEVICE_RECONNECT) {
         // on hot unplug of usb2serial adapters, multiple errors will be
         // reported which is of no importance to the users.
         // reporting it once should be enough
-        QString heading = (m_deviceState == DEVICE_OPENING) ? tr("Error opening device") : tr("Device Error");
-        m_deviceState = DEVICE_CLOSING;
-        QMessageBox::critical(this, heading, m_device->errorString());
-        // this will finally close the device too;
-        controlPanel->closeDevice();
+
+        if ( m_settings->getCurrentSession().autoReconnect ){
+            m_deviceState = DEVICE_RECONNECT;
+            disableInput();
+            if ( m_device->isOpen() ){ //close the device if it was opened but
+                m_device->close();
+            }
+            m_reconnectTimer.start();
+        } else {
+            QString heading = (m_deviceState == DEVICE_OPENING) ? tr("Error opening device") : tr("Device Error");
```

Comment:
The comment about the hot unplug and the error reported only once should be moved to here